### PR TITLE
Add  Alert Table columns 

### PR DESCRIPTION
**Pull request by roshni73**
## Addresses:
- #10 

## Changes
- Add  Alert Table columns event, event category, region, country, admin and view details in the table

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: generated/schema.json
Comment (barshathakuri) [inline generated/schema.json:1]: Please move generated file to gitignore.

File: generated/types.ts
Comment (barshathakuri) [inline generated/types.ts:1]: Please move generated file to gitignore.

File: src/utils/domain/CountryLink/index.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    name: string;
+}
+
+function CountryLink(props: Props) {
+    const { name } = props;
+
+    return (
+        <Link
+            to="/"
+        >
```
Comment (barshathakuri) [inline src/utils/domain/CountryLink/index.tsx]: Add TODO: Add country Link

File: src/utils/domain/RegionLink/index.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+function RegionLink(props: Props) {
+    const { name } = props;
+
+    return (
+        <Link
+            to="/"
+        >
+            {name}
```
Comment (barshathakuri) [inline src/utils/domain/RegionLink/index.tsx]: Add TODO: Add region link

File: src/views/AlertTable/index.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+//   query GetAlerts($filter: AlertFilter) {
+//     alerts(filter: $filter) {
+//       id
+//       eventCategory
+//       region
+//       countries_details
+//       admin
+//     }
+//   }
+// `;
```
Comment (barshathakuri) [inline src/views/AlertTable/index.tsx]: Remove comments

File: src/views/AlertTable/index.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+        id: 2,
+        event: 'Event 2',
+        eventCategory: 'Category 2',
+        region: 'Region 2',
+        countries_details: ['Country 2'],
+        admin: 'Admin 2',
+        sent: false,
+    },
+
+];
```
Comment (barshathakuri) [inline src/views/AlertTable/index.tsx]: Use query data

**Review by barshathakuri — CHANGES_REQUESTED** [review]

File: src/views/AlertTable/i18n.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{
+    "namespace": "emergencies",
+    "strings": {
+        "allOngoingAlertTitle":"ALL ALERTS ",
```
Comment (barshathakuri) [inline src/views/AlertTable/i18n.json]:
```suggestion
        "allOngoingAlertTitle":"All Alerts ",
```
Making it similar to IFRC GO.

File: src/views/AlertTable/index.tsx [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+          name
+        }
+      }
+      alert(pk: "2") {
+        sent
+        url
+      }
+    }
+  }
+`;
```
Comment (barshathakuri) [inline src/views/AlertTable/index.tsx]:
```suggestion
query  AlertType (
   $regionId: ID!,
   $countryId: ID!,
   $alertId: ID!,
    ) {
    public {
      alertInfos {
        items {
          event
          category
        }
      }
      region(pk: $regionId) {
        id
        name
      }
      country(pk: $countryId) {
        id
        name
      }
      admin1s {
        items {
          id
          name
        }
      }
      alert(pk: $alertId) {
        sent
        url
      }
    }
  }
`;
```
Pass the Id in the query.

File: src/views/AlertTable/index.tsx [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+            strings.alertTablecounteries,
+            strings.alertTableAdmins,
+            strings.alertTableSent,
+            strings.alertTableviewDetailsTitle,
+        ],
+    );
+    const {
+        loading,
+        data: alertInfoResponse,
+    } = useQuery<PublicAlertTypeQuery, PublicAlertTypeQueryVariables>(
```
Comment (barshathakuri) [inline src/views/AlertTable/index.tsx]:
```suggestion
    } = useQuery<AlertTypeQuery, AlertTypeQueryVariables>(
```

File: src/views/AlertTable/index.tsx [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+import { useTranslation } from '@ifrc-go/ui/hooks';
+import {
+    createBooleanColumn,
+    createStringColumn,
+} from '@ifrc-go/ui/utils';
+
+import {
+    PublicAlertTypeQuery,
+    PublicAlertTypeQueryVariables,
+} from '#generated/types';
```
Comment (barshathakuri) [inline src/views/AlertTable/index.tsx]:
```suggestion
import {
    AlertTypeQuery,
    AlertTypeQueryVariables,
} from '#generated/types';
```

**Review by barshathakuri — CHANGES_REQUESTED** [review]

File: codegen.yml [hunk 9]
```diff
@@ -14,4 +14,8 @@ generates:
       - "typescript-operations"
   ./generated/schema.json:
     plugins:
+<<<<<<< HEAD
       - "introspection"
+=======
+      - "introspection"
+>>>>>>> 7246516 (Add  Alert Table columns event, event category, region, country, admin and view details in the table)
```
Comment (barshathakuri) [inline codegen.yml]: Remove this

File: generated/types.ts
Comment (barshathakuri) [inline generated/types.ts:1]: Remove this file

File: generated/schema.json
Comment (barshathakuri) [inline generated/schema.json:1]: Remove this file

File: types.stub.ts [hunk 10]
```diff
@@ -1,16 +1,30 @@
...
+<<<<<<< HEAD
 
 export type Mutation = {
   __typename?: 'Mutation';
 };
+=======
+export type Mutation = {
+  __typename?: 'Mutation';
+};
+>>>>>>> 7246516 (Add  Alert Table columns event, event category, region, country, admin and view details in the table)
```
Comment (barshathakuri) [inline types.stub.ts]: Remove <<<< Head. Please rebase it to develop

File: vite.config.ts.timestamp-1712217042372-21fcd90e640fb.mjs
Comment (barshathakuri) [inline vite.config.ts.timestamp-1712217042372-21fcd90e640fb.mjs:1]: Please remove it

**Review by barshathakuri — CHANGES_REQUESTED** [review]

File: src/views/AlertTable/index.tsx [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+    Table,
+} from '@ifrc-go/ui';
+import { SortContext } from '@ifrc-go/ui/contexts';
+import { useTranslation } from '@ifrc-go/ui/hooks';
+import {
+    createDateColumn,
+    createStringColumn,
+} from '@ifrc-go/ui/utils';
+import { isNotDefined } from '@togglecorp/fujs';
+
```
Comment (barshathakuri) [inline src/views/AlertTable/index.tsx:18]: Fix import errors

File: src/views/AlertTable/index.tsx [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+        pageSize: 7,
+        filter: {},
+    });
+
+    const columns = useMemo(
+        () => ([
+            createStringColumn<AlertType, string>(
+                'event',
+                strings.alertTableEvent,
+                (item) => (item.infos.map((info) => info.event).join(', ')),
```
Comment (barshathakuri) [inline src/views/AlertTable/index.tsx]: Fix type issues.

File: package.json [hunk 13]
```diff
@@ -1,13 +1,12 @@
 {
-    "name": "ifrc-alert-hub",
+    "name": "alert-hub",
```
Comment (barshathakuri) [inline package.json]: ifrc-alert-hub

File: src/views/Home/index.tsx [hunk 14]
```diff
@@ -20,9 +30,24 @@ export function Component() {
...
-            />
+            <Tabs
+                value={activeTab}
+                onChange={setActiveTab}
+                variant="secondary"
+            >
+                <TabList>
+                    <Tab name="map"> Map </Tab>
+                    <Tab name="table"> Table </Tab>
+                </TabList>
```
Comment (barshathakuri) [inline src/views/Home/index.tsx]: use translations

**Review by barshathakuri — CHANGES_REQUESTED** [review]

File: src/index.tsx [hunk 15]
```diff
@@ -3,20 +3,30 @@ import 'mapbox-gl/dist/mapbox-gl.css';
...
+} from '@apollo/client';
 
 import App from './App';
 
 const webappRootId = 'webapp-root';
-
 const webappRootElement = document.getElementById(webappRootId);
-
+const client = new ApolloClient({
+    uri: 'http://localhost:8000/graphql/',
```
Comment (samshara) [inline src/index.tsx]: Let's use from env

File: src/views/AlertTable/index.tsx [hunk 16]
```diff
@@ -0,0 +1,193 @@
...
+                (item) => (item.country.name),
+                { sortable: true },
+            ),
+
+            createStringColumn<AlertType, string>(
+                'admin',
+                strings.alertTableAdminsTitle,
+                // NOTE: adim1s may be multiple
+                // so for now we are using first element
+                (item) => item.country.admin1s?.[0]?.name,
```
Comment (samshara) [inline src/views/AlertTable/index.tsx]: Let's concatenate the names for now

File: src/views/AlertTable/index.tsx [hunk 17]
```diff
@@ -0,0 +1,193 @@
...
+            ),
+            createDateColumn<AlertType, string>(
+                'sent',
+                strings.alertTableSentLabel,
+                (item) => (item.sent),
+            ),
+            createLinkColumn<AlertType, string>(
+                'view_details',
+                strings.alertTableViewDetailsTitle,
+                () => 'View Details',
```
Comment (samshara) [inline src/views/AlertTable/index.tsx]: We can use string translations here

**Review by barshathakuri — APPROVED** [review]

**Review by barshathakuri — APPROVED** [review]

**Pull request merged**